adapters/amqp-vertx/src/main/java/org/eclipse/hono/adapter/amqp/impl/VertxBasedAmqpProtocolAdapter.java

```diff
@@ -892,7 +892,20 @@ protected void onCommandReceived(final TenantObject tenantObject, final ProtonSe
         }
 
         // TODO time out waiting for disposition update
+        final Long timerId = vertx.setTimer(getConfig().getDeliveryUpdateTimeout(), tid -> {
+            log.debug("waiting for delivery update timed out after " + getConfig().getDeliveryUpdateTimeout() + " ms");
+            final Exception ex = new ServerErrorException(HttpURLConnection.HTTP_UNAVAILABLE,
+                    "timeout waiting for delivery update");
+            closeLinkWithError(sender, ex, commandContext.getCurrentSpan());
+            // timeout expired -> release command
+            commandContext.getCurrentSpan().log("timeout waiting for delivery update from device");
+            commandContext.release();
+        });
+
         sender.send(msg, delivery -> {
+            // disposition received -> cancel timer
+            vertx.cancelTimer(timerId);
+
             // release the command message when the device either
```

Comment:
A check is needed here whether the timeout has already occurred, skipping the delivery update handling below in that case.